specification/cdn/resource-manager/Microsoft.Cdn/preview/2025-07-01-preview/examples/AFDCustomDomains_Create.json

```diff
@@ -0,0 +1,218 @@
+{
+  "parameters": {
+    "api-version": "2025-07-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "RG",
+    "profileName": "profile1",
+    "customDomainName": "domain1",
+    "customDomain": {
+      "properties": {
+        "hostName": "www.someDomain.net",
+        "tlsSettings": {
+          "certificateType": "ManagedCertificate",
+          "cipherSuiteSetType": "Customized",
+          "minimumTlsVersion": "TLS12",
+          "customizedCipherSuiteSet": {
+            "cipherSuiteSetForTls12": [
+              "ECDHE_RSA_AES128_GCM_SHA256"
+            ],
+            "cipherSuiteSetForTls13": [
+              "TLS_AES_128_GCM_SHA256",
+              "TLS_AES_256_GCM_SHA384"
+            ]
+          }
+        },
+        "mtlsSettings": {
+          "scenario": "ClientCertificateRequiredAndValidated",
+          "secrets": [
+            {
+              "id": "/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/secrets/name1"
+            },
+            {
+              "id": "/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/secrets/name2"
+            }
+          ],
+          "allowedFqdns": [
+            "foo.contoso.com"
+          ]
+        },
+        "azureDnsZone": {
+          "id": ""
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/subid/resourcegroups/RG/providers/Microsoft.Cdn/profiles/profile1/customdomains/domain1",
+        "name": "domain1",
+        "type": "Microsoft.Cdn/profiles/customdomains",
+        "properties": {
+          "profileName": "profile1",
+          "hostName": "www.contoso.com",
+          "tlsSettings": {
+            "certificateType": "ManagedCertificate",
+            "cipherSuiteSetType": "Customized",
+            "minimumTlsVersion": "TLS12",
+            "customizedCipherSuiteSet": {
+              "cipherSuiteSetForTls12": [
+                "ECDHE_RSA_AES128_GCM_SHA256"
+              ],
+              "cipherSuiteSetForTls13": [
+                "TLS_AES_128_GCM_SHA256",
+                "TLS_AES_256_GCM_SHA384"
+              ]
+            },
+            "secret": {
+              "id": ""
+            }
+          },
+          "mtlsSettings": {
+            "scenario": "ClientCertificateRequiredAndValidated",
+            "secrets": [
```

Comment:
>secrets [](http://example.com/codeflow?start=13&length=7)

Is this basically the trusted issuers? Should we call it that? TrustedIssuerSecrets?

Reply:
As written in the description, "secrets" is actually resource references (ie. subs/rg/profile/secret) to Secrets of type MtlsCertificateChain. The "Secret" resource is an entity that we use to group all information regarding keys, certificates, etc. Customers provide us a link to the Azure Key Vault entity storing the actual contents upon creation of the "secret" resource. The "secret" resources are linked to other resources where they are used. For example "mltsSettings" here, and "tlsSettings" too.